ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -48,10 +48,23 @@ def record_error(id, payload, stats: nil)
         end
 
         def record_success(id, stats: nil)
-          redis.pipelined do |pipeline|
+          deleted_count, requeued_count, _ = redis.pipelined do |pipeline|
```

Comment:
[nit] it's not really `deleted_count`, it's more like checking if the test errored before.

So maybe something like

```suggestion
          prev_error_reported, requeued_count, _ = redis.pipelined do |pipeline|
```